tokenizers/src/tokenizer/added_vocabulary.rs

```diff
@@ -304,7 +304,7 @@ impl AddedVocabulary {
             if added_token.single_word {
                 let start_space = start == 0 || sentence.as_bytes().get(start - 1) == Some(&b' ');
```

Comment:
Start and stop are offsets in the strings.

"Hi"

offset 0 -> to the left of H
offset 1 -> between H and i
offset 2 -> right of i

char 0 -> H
char 1 -> i

If the capture is H let's say, then, start=0, offset=1
Then, start == 0 and char 0 = H, so the previous char would indeed be start - 1
And next char  is char 1, so next char == stop_offset indeed (and that's also why it's stop == sentence.len() where it would usually be stop == sentence.len() - 1